src/main/java/apoc/graph/Graphs.java

```diff
@@ -110,4 +112,46 @@ else if (data instanceof Iterator) {
         });
         return Stream.of(new VirtualGraph(name,nodes,rels,props));
     }
+
+    @Description("apoc.graph.fromDocument({json}, {config}) yield graph - transform JSON documents into graph structures")
+    @Procedure(mode = Mode.WRITE)
+    public Stream<VirtualGraph> fromDocument(@Name("json") Object document, @Name(value = "config", defaultValue = "{}") Map<String,Object> config) throws Exception {
+        Collection<Map> coll;
+        if (document instanceof String) {
+            document  = JsonUtil.parse((String) document, null, Object.class);
+        }
+        if (document instanceof Collection) {
+            coll = (Collection) document;
+        } else {
+            coll = Collections.singleton((Map) document);
+        }
+        DocumentToGraph documentToGraph = new DocumentToGraph(db, new GraphsConfig(config));
+        return Stream.of(documentToGraph.create(coll));
+    }
+
+    @Description("apoc.graph.validateDocument({json}, {config}) yield row - validates the json, return the result of the validation")
+    @Procedure(mode = Mode.READ)
+    public Stream<RowResult> validateDocument(@Name("json") String json, @Name(value = "config", defaultValue = "{}") Map<String,Object> config) throws Exception {
+        DocumentToGraph documentGrapher = new DocumentToGraph(db, new GraphsConfig(config));
+        Object jsonObj = JsonUtil.OBJECT_MAPPER.readValue(json, Object.class);
+        Iterator<Map> it;
```

Comment:
why not just a collection and then use a stream.map() for checking and creating the result